examples/src/bin/rtpfecserver.rs

```diff
@@ -0,0 +1,233 @@
+    let q2 = make_element("queue", None)?;
+    let pay = make_element("rtpvp8pay", None)?;
+    let rtpbin = make_element("rtpbin", None)?;
+    let sink = make_element("udpsink", None)?;
+
+    pipeline.add_many(&[&src, &conv, &q1, &enc, &q2, &pay, &rtpbin, &sink])?;
+
+    conv.link(&q1)?;
+    q1.link(&enc)?;
+    enc.link(&pay)?;
+    pay.link(&q2)?;
+
+    rtpbin.connect("request-fec-encoder", false, move |values| {
+        let rtpbin = values[0].get::<gst::Element>().expect("Invalid argument");
+
+        match make_fec_encoder (fec_percentage) {
+            Ok(elem) => Some (elem.to_value()),
+            Err(err) => {
+                gst_element_error!(
+                    rtpbin,
+                    gst::LibraryError::Failed,
+                    ("Failed to make FEC encoder"),
+                    ["{}", err]
+                );
+                None
+            }
+        }
+    })?;
+
+    let srcpad = get_static_pad(&q2, "src")?;
+    let sinkpad = get_request_pad (&rtpbin, "send_rtp_sink_0")?;
+    srcpad.link(&sinkpad).into_result()?;
+
+    let srcpad = get_static_pad(&rtpbin, "send_rtp_src_0")?;
+    let sinkpad = get_static_pad (&sink, "sink")?;
+    srcpad.link(&sinkpad).into_result()?;
+
+    let convclone = conv.clone();
+    src.connect_pad_added(move |decodebin, src_pad| {
+        match connect_decodebin_pad (&src_pad, &convclone) {
+            Ok (_) => (),
+            Err(err) => {
+                gst_element_error!(
+                    decodebin,
+                    gst::LibraryError::Failed,
+                    ("Failed to link decodebin srcpad"),
+                    ["{}", err]
+                );
+                ()
+            }
+        }
+    });
+
+    let video_caps = gst::Caps::new_simple("video/x-raw", &[]);
+
+    src.set_property_from_str("pattern", "ball");
+    sink.set_property("host", &"127.0.0.1".to_value())?;
+    sink.set_property("sync", &true.to_value())?;
+    enc.set_property("keyframe-max-dist", &30i32.to_value())?;
+    enc.set_property("threads", &12i32.to_value())?;
+    enc.set_property("cpu-used", &(-16i32).to_value())?;
+    enc.set_property("deadline", &1i64.to_value())?;
+    enc.set_property_from_str("error-resilient", "default");
+    src.set_property("expose-all-streams", &false.to_value())?;
+    src.set_property("caps", &video_caps.to_value())?;
+    src.set_property("uri", &uri.to_value())?;
+
+    let bus = pipeline.get_bus().expect("Pipeline without bus. Shouldn't happen!");
+
+    let ret = pipeline.set_state(gst::State::Playing);
+    assert_ne!(ret, gst::StateChangeReturn::Failure);
+
+    while let Some(msg) = bus.timed_pop(gst::CLOCK_TIME_NONE) {
+        use gst::MessageView;
+
+        match msg.view() {
+            MessageView::Eos(..) => break,
+            MessageView::Error(err) => {
+                {
+                    match err.get_details() {
+                        Some(details) if details.get_name() == "error-details" => details
```

Comment:
In both applications there is nothing that would ever have Some details. You don't post any error message with details anywhere